### PR TITLE
Support language-specific CSS files #3464

### LINKED ISSUE #3464: Support language-specific CSS files
I wonder if it would be nice to enable language-specific CSS customisations for APIView. Personally I override the styling of Java APIViews with different icons, font sizes, etc, but it would be potentially interesting to enable this for all reviewers of Java APIs, as it may then feel more idiomatic (the current icons do not).

I am happy to supply some Java-specific icons and CSS, if accepted.

### PR DESCRIPTION
**Pull request by cmcd22**
Language specific CSS fields have been created for supported languages. Upon loading a review page, the review language is checked and the appropriate CSS file is linked to.
Fixes #3464 
Java Example: Custom CSS to increase font size and change some font colour to red.
Screen Shot 2022-08-18 at 7 47 05 PM
Java with default CSS:
Screen Shot 2022-08-18 at 7 47 35 PM
C# Example: Custom CSS to decrease font size, change background colour to black and some text colour to white.
Screen Shot 2022-08-18 at 7 48 02 PM
C# with default CSS:
Screen Shot 2022-08-18 at 7 48 38 PM

### CODE REVIEW

**Comment by JonathanGiles** [comment]
@praveenkuttappan @chidozieononiwu Could you please take a look at this PR to support language-specific css functionality?

File: src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -22,6 +22,28 @@
...
     <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-select@1.13.14/dist/css/bootstrap-select.min.css"/>
     <link id="theme-css" rel="stylesheet" href="~/site.css" asp-append-version="true" />
+    @functions {
+        public string getModelType()
+        {
+            return ViewData.ModelMetadata.ModelType.Name;
+        }
+    }
+    @{
+        var modelType = getModelType();
```
Comment (JonathanGiles) [inline src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml:32]: I wonder if it is possible to replace this line as follows (and then drop the `getModelType()` function definition above?
```suggestion
        var modelType = ViewData.ModelMetadata.ModelType.Name;
```

File: src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -22,6 +22,28 @@
...
+        if (modelType == "ReviewPageModel")
+        {
+            @functions {
+                public string getLanguage()
+                {
+                    Assemblies.ReviewPageModel rpm = ViewData.Model;
+                    return rpm.Review.GetLanguageCssSafeName();
```
Comment (JonathanGiles) [inline src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml:39]:
```suggestion
                    return ViewData.Model.Review.GetLanguageCssSafeName();
```

File: src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml (hunk 1 above)
Comment (cmcd22) [reply src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml]: Tested & committed

File: src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml (hunk 2 above)
Comment (cmcd22) [reply src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml]: Tested & committed

**Review by praveenkuttappan — APPROVED** [review]

**Comment by praveenkuttappan** [comment]
You will have to resolve conflict with latest merge from Dozie. Please let me know before you merge after conflict is resolved. I will deploy it to staging and verify before we merge it.

**Comment by JonathanGiles** [comment]
@praveenkuttappan It would be good to deploy this PR to staging along with #3463, as that introduces the custom Java icons.

**Comment by cmcd22** [comment]
@praveenkuttappan conflict resolved

**Comment by JonathanGiles** [comment]
@praveenkuttappan I'm not sure why the build is failing - it seems like it can't find some files, but I'm not sure why?

**Pull request merged**